tests/functional/tests/management/test_add_remove.py

```diff
@@ -313,3 +313,37 @@ def _io_to_core(exported_obj: Core, data: Data):
     completion.wait()
 
     assert completion.results["err"] == 0, "IO to exported object completion"
+
+
+@pytest.mark.parametrize("cache_mode", CacheMode)
+@pytest.mark.parametrize("cls", CacheLineSize)
+def test_load_with_changed_core_size(pyocf_ctx, cache_mode, cls):
+    """
+    Test changing volume size before load
+    :param pyocf_ctx: basic pyocf context fixture
+    :param cm: cache mode we start with
+    :param cls: cache line size we start with
+    """
+    # Start cache device
+    cache_device = Volume(S.from_MiB(50))
+    cache = Cache.start_on_device(
+        cache_device, cache_mode=cache_mode, cache_line_size=cls
+    )
+
+    # Add core to cache
+    core_device = Volume(S.from_MiB(10))
+    core = Core.using_device(core_device)
+    cache.add_core(core)
+    
+    # Stop cache    
+    cache.stop()
+
+    # Change core device size
+    core_device.resize(S.from_MiB(12))
+
+    # Load cache with changed core size
+    with pytest.raises(OcfError, match="OCF_ERR_CORE_SIZE_MISMATCH"):
+        cache_1 = Cache.load_from_device(cache_device)
```

Comment:
please provide "open_cores = False" parameter

Reply:
open_cores parameter is not implemented yet. Test marked with xfail and waiting for merge patches where open_cores parameter is added to load_from_device.